coredb-operator/src/fixtures.rs

```diff
@@ -0,0 +1,150 @@
+//! Helper methods only available for tests
+use crate::{Context, CoreDB, CoreDBSpec, CoreDBStatus, Metrics, COREDB_FINALIZER};
+use assert_json_diff::assert_json_include;
+use futures::pin_mut;
+use http::{Request, Response};
+use hyper::{body::to_bytes, Body};
+use kube::{Client, Resource, ResourceExt};
+use prometheus::Registry;
+use std::sync::Arc;
+use tokio::task::JoinHandle;
+use tower_test::mock::{self, Handle};
+
+impl CoreDB {
+
+    /// A normal test CoreDB
+    pub fn test() -> Self {
+        let mut d = CoreDB::new("testdb", CoreDBSpec::default());
+        d.meta_mut().namespace = Some("testns".into());
+        d.meta_mut().uid = Some("752d59ef-2671-4890-9feb-0097459b18c8".into());
```

Comment:
this line was different than in the example code because we are using ownership reference from CoreDB to StatefulSet, we need the CoreDB to have a uid so this function works: https://docs.rs/kube-core/0.77.0/src/kube_core/resource.rs.html#105

In our create_sts function while it's handling the mocked response.